website/client/components/ui/drawerSlider.vue

```diff
@@ -115,45 +117,58 @@
           next,
         }),
         currentWidth: 0,
-        currentPage: 0,
+        pointer: 0,
       };
     },
     computed: {
-      pages () {
-        return _chunk(this.items, this.itemsPerPage() - 1).map((content, index, array) => {
-          let resultData = [...content];
-
-          if (array[index + 1]) {
-            resultData.push({
-              ...array[index + 1][0],
-              ofNextPage: true,
-            });
-          }
-
-          return resultData;
-        });
+      showItems () {
+        let items = this.items;
+        let pointer = this.pointer;
+        let itemsPerPage = this.itemsPerPage();
+        let firstSlice = items.slice(pointer, pointer + itemsPerPage);
+
+        if (firstSlice.length === itemsPerPage) {
+          return firstSlice;
+        } else {
+          let getRemainderItems = Math.abs(itemsPerPage - firstSlice.length);
+          let secondSlice = items.slice(0, getRemainderItems);
+
+          return firstSlice.concat(secondSlice);
+        }
+      },
+      dividerMargins () {
+        return {
+          marginLeft: `${-this.itemMargin / 2}px`,
+          marginRight: `${this.itemMargin / 2}px`,
+        };
+      },
+    },
+    watch: {
+      itemType () {
+        this.pointer = 0;
       },
     },
     methods: {
-      lastPage () {
-        if (this.currentPage > 0) {
-          this.currentPage--;
+      shiftLeft () {
+        if (this.pointer < this.items.length - 1) {
+          this.pointer++;
         } else {
-          this.currentPage = this.pages.length - 1;
+          this.pointer = 0;
         }
       },
-
-      nextPage () {
-        if (this.currentPage < this.pages.length - 1) {
-          this.currentPage++;
+      shiftRight () {
+        if (this.pointer > 0) {
+          this.pointer--;
         } else {
-          this.currentPage = 0;
+          this.pointer = this.items.length - 1;
         }
       },
-
       itemsPerPage () {
         return Math.floor(this.currentWidth / (this.itemWidth + this.itemMargin));
       },
+      shouldAddVerticalLine (item) {
+        return this.items[this.itemsPerPage() - 1] === item && this.pointer !== 5;
```

Comment:
why the `pointer !== 5` check ?

Reply:
The checker is to not show the pointer as so:

<img width="978" alt="screen shot 2017-12-14 at 4 28 10 pm" src="https://user-images.githubusercontent.com/6742066/34015233-d559c37a-e0eb-11e7-8d66-591fc4cc0952.png">